chaosoci/compute/actions.py

```diff
@@ -1,26 +1,66 @@
 # -*- coding: utf-8 -*-
+from random import choice
+from typing import Any, List, Dict
+
 from oci.core import ComputeClient
+from oci.config import from_file
 
 from chaoslib.types import Configuration, Secrets
+from chaoslib.exceptions import ActivityFailed
 
 from chaosoci import oci_client
 from chaosoci.types import OCIResponse
 
+from .common import get_instances, filter_instances
 
-__all__ = ["stop_instance"]
+__all__ = ["stop_instance", "stop_random_instance"]
 
 
 def stop_instance(instance_id: str, force: bool = False,
                   configuration: Configuration = None,
                   secrets: Secrets = None) -> OCIResponse:
     """Stop a given Compute instance."""
-    action = "SOFTSTOP"
     client = oci_client(ComputeClient, configuration, secrets,
                         skip_deserialization=True)
 
-    if force is True:
-        action = "STOP"
+    action = "STOP" if force else "SOFTSTOP"
+    ret = client.instance_action(instance_id=instance_id, action=action).data
+
+    return ret
+
+
+def stop_random_instance(filters: List[Dict[str, Any]],
+                         compartment_id: str = None,
+                         force: bool = False,
+                         configuration: Configuration = None,
+                         secrets: Secrets = None) -> OCIResponse:
+    """ Stop a a random compute instance within a given compartment.
```

Comment:
I prefer if the docstring starts on a new line:

```
"""
Stop a ...
```

Reply:
Well, still agrees with PEP 257, so I'm not bother in accepting the project's style.